keps/168-pending-workloads-visibility/README.md

```diff
@@ -0,0 +1,314 @@
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+
+As a user of Kueue with LocalQueue visibility I would like to know the
+position of my workload in the local queue. Knowing the position would allow me
+to compute the estimate arrival time (ETA) of my workload.
+
+I would like to be able to get this information by inspecting the local queue
+status.
+
+<!--
+#### Story 2
+
+As a user of Kueue with ClusterQueue visibility I would like to know
+the position of my workload in the cluster queue, so that I can estimate the
+ETA of my workload.
+
+I would like to be able to get this information by inspecting the cluster queue
+status.
+-->
+
+### Notes/Constraints/Caveats (Optional)
+
+<!--
+What are the caveats to the proposal?
+What are some important details that didn't come across above?
+Go in to as much detail as necessary here.
+This might be a good place to talk about core concepts and how they relate.
+-->
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+
+## Design Details
+
+### Local Queue API
+
+```golang
+// PendingWorkload contains the information identifying a pending workload in
+// the local queue.
+type PendingWorkload struct {
+	Name string
+
+	// Namespace indicates the namespace of the pending workload
+	Namespace string
+
+	// Position indicates the position of the workload among all pending
+	// workloads in the cluster queue.
+	Position *int32
+}
+
+type PendingWorkloadsStatus struct {
+	// TopList contains the list of top pending workloads.
+	// +listType=map
+	// +listMapKey=name
+	// +listMapKey=namespace
+	// +optional
+	TopList []PendingWorkload
+}
+
+// LocalQueueStatus defines the observed state of LocalQueue
+type LocalQueueStatus struct {
+...
+	// PendingWorkloadsStatus contains the information exposed about the current
+	// status of the queue of pending workloads.
+	// +optional
+	PendingWorkloadsStatus *PendingWorkloadsStatus
+...
+}
+```
+
+The returned information is controlled by the global Kueue configuration:
+
+```golang
+// Configuration is the Schema for the kueueconfigurations API
+type Configuration struct {
+...
+	// PendingWorkloadsVisibility is configuration to expose the information
+	// about the top pending workloads in the local queue.
+	PendingWorkloadsVisibility *PendingWorkloadsVisibility `json:"pendingWorkloadsVisibility,omitempty"`
+}
+
+type PendingWorkloadsVisibility struct {
```

Comment:
Added now the parameter for the `UpdateBatchPeriod` to control the interval between updates. I will mark this as resolved since we now have a concrete new field